common: no slirp for podman6

lsm5 opened the pull request:
<!--- Please read the [contributing guidelines](https://github.com/containers/container-libs/blob/main/CONTRIBUTING.md) before proceeding --->

See individual commits.

Vendored into https://github.com/containers/podman/pull/28451 and https://github.com/containers/buildah/pull/6443

mtrmac commented:
(I didn’t _read_ the changes yet) it looks like this would benefit from rebasing now that #417 is merged.

lsm5 commented:
Rebased. Good for another look. Thanks.

mtrmac commented on common/pkg/config/config_local_test.go:
```diff
@@ -147,11 +147,6 @@ var _ = Describe("Config Local", func() {
-		config2, err := newLocked(&Options{}, &paths{etc: "testdata/containers_default.conf"})
-		// Then
-		gomega.Expect(err).ToNot(gomega.HaveOccurred())
-		gomega.Expect(config2.Network.DefaultRootlessNetworkCmd).To(gomega.Equal("slirp4netns"))
```
I think we might still want to retain a test that the option can be changed by a config file — just maybe use a `this-is-not-a-valid-command` value?

mtrmac reviewed (commented):
(I have _absolutely no idea_ whether we want to do this or why we are doing this, I’m only looking at the mechanism of the removal.)

lsm5 committed:
rpm: remove slirp4netns weak dependency

The slirp4netns package is no longer needed as Podman 6 only supports
pasta for rootless networking.

Signed-off-by: Lokesh Mandvekar <lsm5@redhat.com>

lsm5 committed:
rpm: remove slirp4netns weak dependency

Signed-off-by: Lokesh Mandvekar <lsm5@redhat.com>

lsm5 committed:
common/pkg/config: remove slirp

Signed-off-by: Lokesh Mandvekar <lsm5@redhat.com>

lsm5 committed:
common/docs: remove slirp

Signed-off-by: Lokesh Mandvekar <lsm5@redhat.com>

lsm5 committed:
common/pkg/ssh: remove slirp

Signed-off-by: Lokesh Mandvekar <lsm5@redhat.com>

lsm5 committed:
common/libnetwork: remove slirp4netns backend, keep RLK helpers

Remove the slirp4netns network backend implementation but retain the
slirp4netns package directory with RootlessKit port-forwarding helper
functions that are still used by podman.

Signed-off-by: Lokesh Mandvekar <lsm5@redhat.com>

lsm5 committed:
common/libnetwork: remove slirp4netns backend, keep RLK helpers

Remove the slirp4netns network backend implementation but retain the
slirp4netns package directory with RootlessKit port-forwarding helper
functions that are still used by podman.

Signed-off-by: Lokesh Mandvekar <lsm5@linux.com>

lsm5 committed:
common/pkg/ssh: remove slirp

Removes the SlirpInfo struct from HostInfo as slirp4netns is no longer
supported in Podman 6.

Signed-off-by: Lokesh Mandvekar <lsm5@redhat.com>

lsm5 committed:
common/pkg/config: remove slirp

Removes the NetworkCmdOptions field (used for slirp4netns arguments) and
updates the test to use a non-slirp test value for config file loading
verification.

Signed-off-by: Lokesh Mandvekar <lsm5@redhat.com>

lsm5 commented:
Opening for review. @containers/container-libs-maintainers PTAL . See description for links to podman and buildah vendor dance PRs. Keeping those in draft until we merge this.

mtrmac commented:
If we remove code from c/common while it is still used in Buildah/Podman, that might make it impossible to vendor container-libs by _others_ working on unrelated features in the meantime, like the config file rework.

Is there any easy way to split/sequence the work to never break vendoring in the meantime, perhaps removing tests+consumers immediately, _then_ removing the called code? (I didn’t check _at all_ whether this is possible/easy.) Or perhaps that already happened?

Luap99 commented:
I agree I think the buildah and podman sides looks safe to merge before the PR here as they only remove the API calls. This would definitely help others doing breaking changes at the same time.

lsm5 commented:
I can redo the buildah and podman PRs without the vendoring changes and defer this one until we're safe to proceed.

lsm5 commented:
switching back to draft.
